gnucash/import-export/ofx/gnc-ofx-import.c

```diff
@@ -1036,6 +1026,101 @@ double ofx_get_investment_amount(const struct OfxTransactionData* data)
     }
 }
 
+// Forward declaration, required because several static functions depend on one-another.
+static void
+gnc_file_ofx_import_int (ofx_info* info);
+
+// gnc_ofx_process_next_file processes the next file in the info->file_iter list.
+static void
+gnc_ofx_process_next_file (GtkDialog *dialog, gpointer user_data)
+{
+    ofx_info* info = (ofx_info*) user_data;
+    if (info->file_iter->next)
+    {
+        // Done with the previous OFX file, process the next one if any.
+        info->file_iter = info->file_iter->next;
+        gnc_file_ofx_import_int (info);
+    }
+    else
+    {
+        // No more OFX file to process, final cleanup.
+        g_slist_free_full (info->file_list, g_free);
+        g_free (info);
```

Comment:
This seems more logical to be added at the end of gnc_file_ofx_import

Reply:
mmm, all the calls in gnc_file_ofx_import() are non-blocking so the end of the function is reached instantly before info is done being used. So you can't free it there. You have to free it after the last time it's used, correct?

Reply:
That's OK then, I did not see that but yes it is best to free them when no longer required.